osquery/core/windows/wmi.cpp

```diff
@@ -319,14 +478,10 @@ WmiRequest::WmiRequest(const std::string& query, BSTR nspace) {
 
     hr = enum_->Next(WBEM_INFINITE, 1, &result, &result_count);
     if (SUCCEEDED(hr) && result_count > 0) {
-      results_.push_back(WmiResultItem(result));
+      results_.emplace_back(WmiResultItem(result, locator_, services_));
```

Comment:
```suggestion
      results_.emplace_back(result, locator_, services_);
```
Otherwise it still builds a temporary and copy/moves it.